autometa/config/databases.py

```diff
@@ -66,230 +97,504 @@ class Databases:
+        """Get the checksum from provided `section` respective to `option` in `self.config`.
+
+        Parameters
+        ----------
+        section : str
+            section to retrieve for `checksums` section.
+            Choices include: 'ncbi' and 'markers'.
+        option : str
+            `option` in `checksums` section corresponding to the section checksum file.
+
+        Returns
+        -------
+        str
+            checksum of remote md5 file. e.g. 'hash filename\n'
+
+        Raises
+        -------
+        ConnectionError
+            Failed to connect to host for provided `option`.
+
+        """
+        if section not in {'ncbi','markers'}:
+            raise ValueError(f'"section" must be "ncbi" or "markers". Provided: {section}')
+        if section == 'ncbi':
+            host = self.config.get(section, 'host')
+            ftp_fullpath = self.config.get('checksums', option)
+            chksum_fpath = ftp_fullpath.split(host)[-1]
+            with FTP(host) as ftp, tempfile.TemporaryFile() as fp:
+                ftp.login()
+                result = ftp.retrbinary(f'RETR {chksum_fpath}', fp.write)
+                if not result.startswith('226 Transfer complete'):
+                    raise ConnectionError(f'{chksum_fpath} download failed')
+                ftp.quit()
+                fp.seek(0)
+                checksum = fp.read().decode()
+        elif section == 'markers':
+            url = self.config.get('checksums', option)
+            with requests.Session() as session:
+                resp = session.get(url)
+                if not resp.ok:
+                    raise ConnectionError(f'Failed to retrieve {url}')
+                checksum = resp.text
+        return checksum
 
     def format_nr(self):
-        """Format NCBI nr.gz database into diamond formatted database nr.dmnd.
+        """Construct a diamond formatted database (nr.dmnd) from `nr` option
+        in `ncbi` section in user config.
+
+        NOTE: The checksum 'nr.gz.md5' will only be generated if nr.dmnd
+        construction is successful. If the provided `nr` option in `ncbi` is 'nr.gz'
+        the database will be removed after successful database formatting.
 
         Returns
         -------
         NoneType
             config updated option:'nr' in section:'ncbi'.
 
         """
-        db_infpath = self.config.get('ncbi', 'nr')
-        db_outfpath = db_infpath.replace('.gz', '.dmnd')
-        if not self.dryrun and not os.path.exists(db_outfpath):
-            diamond.makedatabase(
-                fasta=nr, database=db_infpath, nproc=self.nproc)
+        db_infpath = self.config.get('ncbi','nr')
+        db_infpath_md5 = f'{db_infpath}.md5'
+        db_outfpath = db_infpath.replace('.gz','.dmnd')
+        checksums_match = False
+        if os.path.exists(db_infpath_md5) and os.path.exists(db_outfpath):
+            # If nr.dmnd.md5 exists, then we will check if nr.gz.md5 is matching
```

Comment:
See my comment above. The naming of these files is very confusing. Shouldn't it be `nr.dmnd.md5`?

Reply:
The reason I'm only doing this for `nr.gz.md5` is because we have a reference md5 for `nr.gz` but we do not have a reference for `nr.dmnd.md5` a priori.

I think this is a subtle difference that may be more specific for issue-#71? I'm not sure

Reply:
Yes, perhaps. I just think that since the conversion to a .dmnd file takes a long time, it is important here to get a checksum at the end (only provided, of course, that the diamond command finished without error).

Reply:
After thinking over the checksum writing this weekend. I agree with you on writing checksums after each successful download of each of the database files. As well as writing checksums for formatted files like `nr.dmnd` that take a long time to create. I'm not sure we need to checksum the formatted markers or the files within taxdump. Although maybe we write a database summary file and add in the checksums from the downloaded and formatted databases? So in this case we would calculate checksums for _all_ of the formatted files and write them to one database summary file. If any of the corresponding database checksums are updated, this would be reflected in the summary checksum. I'm thinking of logic similar to a Makefile here. This way we can update the formatted databases (`*.h3*`, `nr.dmnd`, `nodes.dmp`, `names.dmp`, `merged.dmp`) when any of their respective parent database files are updated while also ensuring file integrity later with lookup of the database summary during any Autometa run.

Reply:
Yes, I think a single file for checksums would work. My argument for doing a checksum for every output file is that it is much more work devising all sorts of tests for the format of each file to check it is OK and not corrupted. Checksums are both (1) foolproof/accurate and (2) easy.

Reply:
Just looked at the code changes in this function, and they look good. The one problem is that we are now checking that the `nr.dmnd.md5` file exists, but not that the checksum matches that of the file as it now exists on the disk.